tgui/packages/tgui/components/Dropdown.tsx

```diff
@@ -0,0 +1,297 @@
+import { createPopper, VirtualElement } from '@popperjs/core';
+import { classes } from 'common/react';
+import { Component, findDOMfromVNode, InfernoNode, render } from 'inferno';
+import { Box, BoxProps } from './Box';
+import { Icon } from './Icon';
+
+export interface DropdownEntry {
+  displayText: string | number | InfernoNode;
+  value: string | number | Enumerator;
+}
+
+type DropdownUniqueProps = {
+  options: string[] | DropdownEntry[];
+  icon?: string;
+  iconRotation?: number;
+  clipSelectedText?: boolean;
+  width?: string;
+  menuWidth?: string;
+  over?: boolean;
+  color?: string;
+  nochevron?: boolean;
+  displayText?: string | number | InfernoNode;
+  onClick?: (event) => void;
+  // you freaks really are just doing anything with this shit
+  selected?: any;
+  onSelected?: (selected: any) => void;
+};
+
+export type DropdownProps = BoxProps & DropdownUniqueProps;
+
+const DEFAULT_OPTIONS = {
+  placement: 'left-start',
+  modifiers: [
+    {
+      name: 'eventListeners',
+      enabled: false,
+    },
+  ],
+};
+const NULL_RECT: DOMRect = {
+  width: 0,
+  height: 0,
+  top: 0,
+  right: 0,
+  bottom: 0,
+  left: 0,
+  x: 0,
+  y: 0,
+  toJSON: () => null,
+};
```

Comment:
```suggestion
} as const;
```
Good idea unless it gives type errors elsewhere